guides/source/testing.md

```diff
@@ -859,292 +846,440 @@ users(:david, :steve)
 Model Testing
 -------------
 
-Model tests are used to test the various models of your application.
+Model tests are used to test the various models of your application and their
+associated logic. You can test this logic using the assertions and fixtures that
+we've explored in the sections above.
 
-Rails model tests are stored under the `test/models` directory. Rails provides
-a generator to create a model test skeleton for you.
+Rails model tests are stored under the `test/models` directory. Rails provides a
+generator to create a model test skeleton for you.
 
 ```bash
 $ bin/rails generate test_unit:model article title:string body:text
 create  test/models/article_test.rb
-create  test/fixtures/articles.yml
 ```
 
-Model tests don't have their own superclass like `ActionMailer::TestCase`. Instead, they inherit from [`ActiveSupport::TestCase`](https://api.rubyonrails.org/classes/ActiveSupport/TestCase.html).
+Model tests don't have their own superclass like `ActionMailer::TestCase`.
+Instead, they inherit from
+[`ActiveSupport::TestCase`](https://api.rubyonrails.org/classes/ActiveSupport/TestCase.html).
 
-System Testing
---------------
+Functional Testing for Controllers
+----------------------------------
 
-System tests allow you to test user interactions with your application, running tests
-in either a real or a headless browser. System tests use Capybara under the hood.
+In Rails, a focus on testing the various actions of a controller is a form of
+writing functional tests. Remember, your controllers handle the incoming web
```

Comment:
For me this first sentence reads a bit awkward, it seems to be phrased in the wrong order.
Maybe it's clearer if we start with the "functional tests":
```suggestion
In Rails, functional tests focus on testing the various actions of a controller.
Remember, your controllers handle the incoming web
```

Or maybe remove this paragraph altogether? The next paragraph seems to repeat most of it.